core/templates/pages/exploration-player-page/services/player-position.service.ts

```diff
@@ -55,6 +56,9 @@ export class PlayerPositionService {
    * @return {string} a string that shows the name of current state.
    */
   getCurrentStateName(): string {
+    if (this.displayedCardIndex === null) {
+      throw new Error('Displayed Card Index has not been set');
```

Comment:
This is a getter, I think we should cater these errors at places where this function is used .